test/runtime_tests.cc

```diff
@@ -1013,5 +1015,70 @@ namespace detail {
 		dry_run_with_nodes(nodes);
 	}
 
+	TEST_CASE_METHOD(test_utils::mpi_fixture, "Config reads environment variables correctly", "[env-vars]") {
+		std::unordered_map<std::string, std::string> valid_test_env_vars{
+		    {"CELERITY_LOG_LEVEL", "debug"},
+		    {"CELERITY_GRAPH_PRINT_MAX_VERTS", "1"},
+		    {"CELERITY_DEVICES", "0 0"},
+		    {"CELERITY_PROFILE_KERNEL", "1"},
+		    {"CELERITY_DRY_RUN_NODES", "4"},
+		};
+
+		const auto test_env = env::scoped_test_environment(valid_test_env_vars);
+		auto cfg = config(nullptr, nullptr);
+
+		REQUIRE(spdlog::get_level() == spdlog::level::debug);
```

Comment:
Prefer `CHECK` over `REQUIRE` unless the code afterwards would crash (e.g. if there's fewer than expected elements in array).